cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -467,7 +467,15 @@ ObjectId resolveHead() throws IOException {
       throw new IllegalStateException(
           "Call cloneRemote() before getting HEAD.");
     }
-    return git.getRepository().resolve(Constants.HEAD);
+
+    ObjectId commitObjectId = git.getRepository().resolve(Constants.HEAD);
+
+    if (commitObjectId == null) {
+      throw new SourceControlException(
```

Comment:
Need to update java doc with the new exception.

Reply:
updated